classes/class-p4-context.php

```diff
@@ -59,10 +59,32 @@ public static function set_og_meta_fields( &$context, $post ) {
 	public static function set_campaign_datalayer( &$context, $meta ) {
 		$context['cf_campaign_name'] = $meta['p4_campaign_name'] ?? '';
 		$context['cf_basket_name']   = $meta['p4_basket_name'] ?? '';
-		$context['cf_scope']         = $meta['p4_scope'] ?? '';
 		$context['cf_department']    = $meta['p4_department'] ?? '';
 		$context['cf_project_id']    = $meta['p4_global_project_tracking_id'] ?? 'not set';
 		$context['cf_local_project'] = $meta['p4_local_project'] ?? 'not set';
+		$context['cf_scope']         = self::get_campaign_scope( $context['cf_campaign_name'], $context['cf_local_project'] );
 	}
 
+	/**
+	 * Get campaign scope from value selected in the Global Projects dropdown.
+	 * Conditions:
+	 * - If Global Project equals Local Campaign then Scope is Local.
+	 * - If Global Project equals none then Scope is not set
+	 * - If Global Project matches any other value (apart from Local Campaign) then Scope is Global
+	 *
+	 * @param string $global_project The Global Project value.
+	 * @param string $local_project The Local Project value.
+	 * @return string The campaign scope.
+	 */
+	private static function get_campaign_scope( $global_project, $local_project ) {
+		if ( 'not set' === $global_project ) {
```

Comment:
Maybe this can be a switch as both if's are checking the same variable. You could also directly return for each case, so you don't nee the `$campaign_scope` variable or break statement.